compiler/testData/diagnostics/nativeTests/sharedImmutable.kt

```diff
@@ -0,0 +1,139 @@
+// FILE: annotation.kt
+package kotlin.native.concurrent
+
+import kotlin.reflect.KProperty
+
+@Target(AnnotationTarget.PROPERTY)
+@Retention(AnnotationRetention.BINARY)
+annotation class SharedImmutable
+
+@Target(AnnotationTarget.PROPERTY, AnnotationTarget.CLASS)
+@Retention(AnnotationRetention.BINARY)
+annotation class ThreadLocal
+
+// FILE: test.kt
+import kotlin.native.concurrent.SharedImmutable
+import kotlin.native.concurrent.ThreadLocal
+import kotlin.reflect.KProperty
+
+fun println(<!UNUSED_PARAMETER!>value<!>: Int) {}
+fun println(<!UNUSED_PARAMETER!>value<!>: String) {}
+fun println(<!UNUSED_PARAMETER!>value<!>: Point) {}
+
+data class Point(val x: Double, val y: Double)
+@SharedImmutable
+val point1 = Point(1.0, 1.0)
+
+<!INAPPLICABLE_SHARED_IMMUTABLE_PROPERTY!>@SharedImmutable<!>
+var point2 = Point(2.0, 2.0)
+
+class Date(<!INAPPLICABLE_SHARED_IMMUTABLE_TOP_LEVEL!>@SharedImmutable<!> val month: Int, <!INAPPLICABLE_SHARED_IMMUTABLE_PROPERTY, INAPPLICABLE_SHARED_IMMUTABLE_TOP_LEVEL!>@SharedImmutable<!> var day:Int)
+class Person(val name: String) {
+    <!INAPPLICABLE_SHARED_IMMUTABLE_PROPERTY, INAPPLICABLE_SHARED_IMMUTABLE_TOP_LEVEL!>@SharedImmutable<!>
+    var surname: String? = null
+}
+
+class Figure {
+    <!INAPPLICABLE_SHARED_IMMUTABLE_PROPERTY, INAPPLICABLE_SHARED_IMMUTABLE_TOP_LEVEL!>@SharedImmutable<!>
+    val cornerPoint: Point
+        get() = point1
+}
+
+<!INAPPLICABLE_SHARED_IMMUTABLE_PROPERTY!>@SharedImmutable<!>
+var age = 20
+get() {
+    println("Age is: $field")
+    return field
+}
+set(value) {
+    println(value)
+}
+
+var globalAge = 30
+<!INAPPLICABLE_SHARED_IMMUTABLE_PROPERTY!>@SharedImmutable<!>
+var age1 = 20
+    get() {
+        println("Age is: $field")
+        return field
+    }
+    set(value) {
+        globalAge = value
+    }
+
+// Can't
+<!INAPPLICABLE_SHARED_IMMUTABLE_PROPERTY!>@SharedImmutable<!>
+var point3: Point
+    get() = point2
+    set(value) {
+        point2 = value
+    }
+
+<!INAPPLICABLE_SHARED_IMMUTABLE_PROPERTY!>@SharedImmutable<!>
+var point4: Point
+    get() = point2
+    set(value) {
+        println(value)
+    }
+
+@ThreadLocal
+var point0 = Point(2.0, 2.0)
```

Comment:
Also `@SharedImmutable @ThreadLocal` should probably be forbidden.

Reply:
Decided to do separately later.